### PR TITLE
Improve logging for features and their state

### PR DESCRIPTION
**Pull request by pierDipi**
This would help debug test failures.

### CODE REVIEW

File: pkg/feature/feature.go [hunk 1]
```diff
@@ -41,6 +42,29 @@ type Feature struct {
...
+	}{
+		Name:  f.Name,
+		Steps: f.Steps,
+		State: f.State,
+		Refs:  f.refs,
+	}
+	return json.MarshalIndent(in, "", " ")
+}
+
+func (f Feature) Logf(log func(args ...interface{})) {
```
Comment (devguyio) [inline pkg/feature/feature.go:60]: - The name was very confusing for me, I thought it'd be a `Formatted log of its args` . This looks more like "dump feature state using log function". Wdyt? maybe `f.DumpWith(T.Log)` or something like that
- NIT: method doc?

**Review by devguyio — COMMENTED** [review]
nice idea! left a comment

File: pkg/feature/feature.go (hunk 1 above)
Comment (pierDipi) [reply pkg/feature/feature.go]: Done

**Review by devguyio — APPROVED** [review]
/lgtm

**Pull request merged**